homeassistant/components/awair/config_flow.py

```diff
@@ -21,20 +23,79 @@ class AwairFlowHandler(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    _device: AwairLocalDevice
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+
+        host = discovery_info.host
+        LOGGER.debug("Discovered device: %s", host)
+
+        self._device, _ = await self._check_local_connection(host)
+
+        if self._device is not None:
+            await self.async_set_unique_id(self._device.mac_address)
+            self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+            self.context.update(
+                {
+                    "title_placeholders": {
+                        "host": host,
+                        "model": self._device.model,
+                    }
+                }
+            )
+        else:
+            return self.async_abort(reason="unreachable")
```

Comment:
Can we differentiate between the device not being reachable versus the device is an Awair device but the local API is disabled? Because in the latter case we shouldn't abort, but instead guide the user on how to enable their local API.

Reply:
This one is still open. Let's address this in a future PR. My Awair Element is coming in on Friday so I can help improve this experience.

Reply:
Sure, makes sense improve it in the future. FWIW, the API is exposed on `:80` and the device does not listen on this port until the API is enabled. Perhaps ICMP can be used to check recachability?

Reply:
Does the device only broadcast mDNS when the API is enabled?

Reply:
That is correct, mDNS is only broadcast when the API is enabled. That being said, hass receiving a mDNS packet doesn't necessarily mean that hass will be able to connect back to the awair device. This happened in my environment initially where I allowed mDNS broadcasts across VLAN, but firewall was preventing the tcp connection.